enhancements/network/ovn-kubernetes-evpn.md

```diff
@@ -0,0 +1,429 @@
+  MetalLB when manually configured to be advertised through EVPN.
+
+QE testing will need to emulate a BGP/EVPN fabric. While any kind of custom and
+simplified setup is acceptable, there might be interest in using third party
+projects like [containerlab][14] which facilitate the orchestration of an
+emulated fabric with several different implementations, including FRR.
+
+[14]: https://github.com/srl-labs/containerlab
+
+## Graduation Criteria
+
+The EVPN feature is planned to be provided with technical preview availability
+first and then with general availability in a later release. While the main
+development effort will take place in context of upstream projects, the
+following graduation criteria includes suggestions to prioritize that
+development effort in case it can be accommodated.
+
+### Dev Preview -> Tech Preview
+
+- EVPN support for L2 and L3 P-CUDNs
+- Use of raw configuration API in FRR-k8s
+- Use of node IPs as VTEP IPs
+- Sufficient test coverage (E2E, QE)
+
+### Tech Preview -> GA
+
+- Formal FRR-k8s APIs introduced
+- Configurable VTEP IPs
+- Complete testing, including upgrades
+- User facing documentation available in [openshift-docs][15]
+
+[15]: https://github.com/openshift/openshift-docs/
+
+### Removing a deprecated feature
+
+NA
+
+## Upgrade / Downgrade Strategy
+
+This feature has no impacts on upgradeability.
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+N/A
+
+## Support Procedures
+
+In general, support procedures will be based on the status reported on the
+resource instances involved in the EVPN feature which are similar to those
+already involved in the existing BGP features, including the status for
+`RouteAdvertisements`, `ClusterUserDefinedNetwork` and FRR-k8s resources: any
+invalid configuration shall be reported through appropriate status conditions
+and events associated to those resources.
+
+Status conditions should have metrics associated, allowing the configuration of
+alerts based on those metrics. FRR-k8s is expected to provide additional metrics
+equivalent to those already provided for the [unicast address family][16]
+
+[16]: https://github.com/openshift/frr/blob/release-4.22/frr-tools/metrics/collector/metrics.go
+
+Other than that, and given the distributed nature of OVN-Kubernetes, the next
+best troubleshooting method relies on the use tools like `iproute2`, `tcpdump`,
+`ovn-trace`, `ovs-trace` and `vtysh`, mixing existing knowledge with additional
+understanding of FRR and host configuration specific to EVPN that is detailed in
+the upstream [enhancement][17]
+
+[17]: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/4568650412f85cdcf7671c74d2e9b3bb67d3b366/docs/okeps/okep-5088-evpn.md#implementation-details
+
+In the future, NetObserv might introduce insights into that host configuration
+and facilitate troubleshooting.  
+
+## Infrastructure Needed [optional]
+
+The EVPN feature is only supported for baremetal platforms.
```

Comment:
not sure if the template didn't have a perf/scale section but maybe good to add one for our own accountability sake calling out any expected complications we see for perf/scale - if none then that's good to be stated.

Reply:
Added section